enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,429 @@
+  │    │     │        │ External Switch │                   │    │
+  │    │     │        └────────┬────────┘                   │    │
+  │    │     │                 │                            │    │
+  │    │     │     ┌───────────┴──────────┐                 │    │
+  │    │     │     │     Gateway Router   │                 │    │
+  │    │     │     └───────────┬──────────┘                 │    │
+  │    │     │                 │                            │    │
+  │    │     │         ┌───────┴────────┐                   │    │
+  │    │     │         │  Join Switch   │                   │    │
+  │    │     │         └───────┬────────┘                   │    │
+  │    │     │                 │                            │    │
+  │    │     │   ┌─────────────┴─────────────┐              │    │
+  │    │     │   │    OVN Cluster Router     │              │    │
+  │    │     │   │        10.244.0.1/24      │              │    │
+  │    │     │   └─────────────┬─────────────┘              │    │
+  │    │     │                 │                            │    │
+  │    │     │        ┌────────┴─────────┐                  │    │
+  │    │     │        │    Node Switch   │                  │    │
+  │    │     │        └────────┬─────────┘                  │    │
+  │    │     │                 │                            │    │
+  │    │  ┌──┴────────────┐    │      ┌──────────────┐      │    │
+  │    │  │ mp0           ├────┴──────┤Pod           │      │    │
+  │    │  │ 10.244.0.2/24 │           │10.244.0.50/24│      │    │
+  │    │  └───────────────┘           └──────────────┘      │    │
+  │    │                                                    │    │
+  │    └────────────────────────────────────────────────────┘    │
+  │                                                              │
+  └──────────────────────────────────────────────────────────────┘
+```
+
+There will be no change to the overlay path taken by the east-west traffic
+between the pods. 
+
+The following north-south traffic paths will not be affected:
+
+* Overlay pod to external
+* Host to ClusterIP services backed by overlay pods
+* Host to NodePort services backed by overlay pods
+
+The path for following north-south traffic case will be slightly different:
+
+* Host to ClusterIP services backed by host pods
+* Host to NodePort services backed by host pods
+* External to NodePort services
+
+### Host to ClusterIP Services Backed by Host Pods
+
+The service and endpoint are:
+
+```bash
+[root@rhel92 microshift]# oc get svc nginx-host
+NAME         TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)        AGE
+nginx-host   NodePort    10.43.117.48   <none>        80:30847/TCP   5d21h
+[root@rhel92 microshift]# oc get ep nginx-host
+NAME         ENDPOINTS            AGE
+nginx-host   10.89.0.3:8080       5d21h
+```
+
+The general flow is:
+1. The packet is sent to the OVN gateway router via br-ex with src_ip (10.44.0.1),
```

Comment:
> Have you tracked down how OVN-K uses Node status Internal IPs vs Node annotated IP addresses (which I guess are different here?) and what features might work differently or not work because of this?

OVN-K thinks the IP (10.44.0.1) of br-ex is the local host IP.  And OVN-K treats the NodeIP(10.89.0.3) as an external IP. It means the endpoint IP of services backed by hostNetwork pods is NodeIP. Therefore those traffic cases are affected.

The only broken feature is the service with `internalTrafficPolicy=Local`. The traffic cases I depict in this document are the ones that work differently.

Reply:
Is this for example affecting egress ip feature? It uses the node's IP addresses to avoid SNAT'ing cluster-local traffic. See `CreateDefaultNoRerouteNodePolicies`. 

I guess that 10.44.0.1 wouldn't be used as an endpoint unless a workload explicitly uses it but that is still a possibility.

Should we track down if there are other features doing something similar that might be affected?

The real fix might be for these features to not rely in the node status IP, but instead on the annotated IPs.

Reply:
@jcaamano and @pliurh , is this thread ready to be resolved?

Reply:
I updated the `Drawbacks` chapter by adding the features which will be affected. PTAL.